tests/scripts/thread-cert/border_router/test_publish_meshcop_service.py

```diff
@@ -130,6 +130,12 @@ def test(self):
         self.check_meshcop_service(br1, host)
         self.check_meshcop_service(br2, host)
 
+        br1.factory_reset()
+        br1.set_network_name('ot-br-1-3')
```

Comment:
What's the purpose of setting a new network name here?

Reply:
This test cases uses network name to distinguish the meshcop services. Here we set a new network name to ensure that the discovered meshcop services are those we expect.

Reply:
The BR should already have published the meshcop service before set_network_name. Would `set_network_name` cause it to re-publish?

Reply:
Yes.